src/application/pets.ts

```diff
@@ -2,9 +2,12 @@ import {
   AdoptablePets, GetLocation, GetPets, IPAddress, isIPAddress, Location, Pet
 } from '../domain/model'
 
-export type Env = GetLocation<IPAddress, Location | null> & GetPets<Location, readonly Pet[]>
+export type Env<LocationEnv, PetsEnv> = LocationEnv & PetsEnv & {
+  getLocation: GetLocation<LocationEnv, IPAddress, Location | null>,
+  getPets: GetPets<PetsEnv, Location, readonly Pet[]>
+}
```

Comment:
For this specific example, my preference is towards the new change.  When looking at an `Env`, it's fewer surprises of what could be contained in the `GetPets` object.  I also like how it reads if you need to mix and match infrastructure

Example,
```
export type Env = {
  getLocation: IPStack.GetLocation,
  getPets: Brushfire.GetPets,
  getApplicants: AdoptionService.GetApplicants
}
```

Reply:
> it's fewer surprises of what could be contained in the GetPets object

Hmmm, can you say more about ways you might be surprised, given that the GetPets object type is pretty easy to observe?

> I also like how it reads if you need to mix and match infrastructure

Yeah, I think this is an interesting tradeoff.  It gives the consumer more control over names and name spacing (by constructing their own Env type that controls names).  On the other hand, object interfaces are easily composable via intersection, but you give up control of the names.

Reply:
I think surprise by `GetPets` is the wrong phrase, I think my concern is more surprise as to what's in the `Env`

I think for simple cases of this union, `Env = Location & Pets`, it is easy to observe.  

When you have something like, `Env = Location & Pets & Owners & Shelters & Volunteers & Vets`, at some point I'll look at some generic field like `Env.fetch(...)` and wonder where it's coming from.  Is it being overridden by namespace conflicts?